src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -94,96 +114,169 @@ pub enum FilesystemEventKind {
 // ==================
 
 /// Attributes of the file in the filesystem.
-#[derive(Clone,Copy,Debug,PartialEq)]
+#[derive(Clone,Debug,PartialEq,Eq,Hash)]
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Attributes{
     /// When the file was created.
-    pub creation_time      : UTCDateTime,
+    pub creation_time : UTCDateTime,
     /// When the file was last accessed.
-    pub last_access_time   : UTCDateTime,
+    pub last_access_time : UTCDateTime,
     /// When the file was last modified.
     pub last_modified_time : UTCDateTime,
     /// What kind of file is this.
-    pub file_kind          : FileKind,
+    pub kind : FileSystemObject,
     /// Size of the file in bytes.
     /// (size of files not being `RegularFile`s is unspecified).
-    pub byte_size          : u64
+    pub byte_size : u64
 }
 
-/// What kind of file (regular, directory, symlink) is this.
-#[derive(Clone,Copy,Debug,PartialEq)]
-#[derive(Serialize, Deserialize)]
-pub enum FileKind {
-    /// File being a directory.
-    Directory,
-    /// File being a symbolic link.
-    SymbolicLink,
-    /// File being a regular file with opaque content.
-    RegularFile,
-    /// File being none of the above, e.g. a physical device or a pipe.
-    Other
+/// A representation of what kind of type a filesystem object can be.
+#[derive(Hash,Debug,Clone,PartialEq,Eq,Serialize,Deserialize)]
+#[serde(tag = "type")]
+pub enum FileSystemObject {
```

Comment:
All these things should be in a module named `file::system::`. Plus this should be renam,ed from `Object` to `Node` as "file system node" is a popular term for that

Reply:
@farmaazon :

1. If the name is dictated by a protocol, then all such things should be in a common module, like `engine::protocol::api`
2. Its not the responsibility of the reviewer to ask engine for a change ! This is the responsibility of people connecting to the API. So if we will come to a conclusion here to change it, it is the responsibility of Danilo or you as the team leader. Definitely not someone who tells that the code is ugly now.


Regarding: 
```
The module file::system would suggest, that we're implementing filesystem operations, but we just implement the client for an external service. 
```

I completely agree. Now it seems like we are implementing filesystem operations. If this is a protocol for engine, move it to a place where it looks like engine protocol, like `engine::protocol::api`

Reply:
@wdanilo but this is engo-protocol crate! Is it not sufficient to explain that those are clients using this API. But anyway, the information about "keeping names as in protocol" should be put in the module docs.

2. Yes it is. If you have concerns about names used in engine protocil, you're making yourself a reviewer of engine protocol. And the reviewer of engine protocol should request for changes in engine protocol, not a random developer who is using this API.

Reply:
`File System` is a concept. A module named `file_system` would be fine for me. But I agree that things should follow as much as possible the protocol documentation.